core/services/relay/evm/dual_contract_transmitter.go

```diff
@@ -173,8 +177,67 @@ func (oc *dualContractTransmitter) FromAccount(ctx context.Context) (ocrtypes.Ac
 	return ocrtypes.Account(oc.transmitter.FromAddress(ctx).String()), nil
 }
 
-func (oc *dualContractTransmitter) Start(ctx context.Context) error { return nil }
-func (oc *dualContractTransmitter) Close() error                    { return nil }
+func (oc *dualContractTransmitter) lockTransmitters(ctx context.Context) error {
```

Comment:
I suggest push locking to the end of the function to make it explicit that either you acquire both locks or none in this function.
```suggestion
func (oc *dualContractTransmitter) lockTransmitters(ctx context.Context) error {
	primaryAddress := oc.transmitter.FromAddress(ctx)
	rmPrimary, err := oc.ks.GetResourceMutex(ctx, primaryAddress)
	if err != nil {
		return err
	}

	secondaryAddress, err := oc.transmitter.SecondaryFromAddress(ctx)
	if err != nil {
		return err
	}
	rmSecondary, err := oc.ks.GetResourceMutex(ctx, secondaryAddress)
	if err != nil {
		return err
	}

	errPrimary := rmPrimary.TryLock(keystore.TXMv1)
	if errPrimary != nil {
		return errPrimary
	}

	errSecondary := rmSecondary.TryLock(keystore.TXMv2)
	if errSecondary != nil {
		return multierr.Append(err, rmPrimary.Unlock(keystore.TXMv1)
	}

	oc.lggr.Debugf("Keys %s and %s has been locked for TXMv1 and TXMv2", primaryAddress.String(), secondaryAddress.String())
	return nil
}
```

Reply:
Implemented in [`140aa03` (#15969)](https://github.com/smartcontractkit/chainlink/pull/15969/commits/140aa03deae1d1d0b57bb60678433b5b950019c8)